src/ray/core_worker/core_worker.cc

```diff
@@ -1955,16 +1984,27 @@ Status CoreWorker::KillActor(const ActorID &actor_id, bool force_kill, bool no_r
   if (options_.is_local_mode) {
     return KillActorLocalMode(actor_id);
   }
-
-  if (!actor_manager_->CheckActorHandleExists(actor_id)) {
-    std::stringstream stream;
-    stream << "Failed to find a corresponding actor handle for " << actor_id;
-    return Status::Invalid(stream.str());
-  }
-
-  RAY_CHECK_OK(
-      gcs_client_->Actors().AsyncKillActor(actor_id, force_kill, no_restart, nullptr));
-  return Status::OK();
+  std::promise<Status> p;
```

Comment:
This is the new code for the fixing.

Reply:
What was the root cause of the issue? Sounds like thread-safety?

Reply:
No. It's just a race condition. It's trying to kill the actor that's pending registration.

Reply:
Got it.